specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-06-01/managedClusters.json

```diff
@@ -2780,6 +2786,11 @@
         "creationData": {
           "$ref": "#/definitions/CreationData",
           "description": "CreationData to be used to specify the source Snapshot ID if the node pool will be created/upgraded using a snapshot."
+        },
+        "hostGroupID": {
+          "type": "string",
+          "title": "The fully qualified resource ID of the Dedicated Host Group to provision virtual machines from, used only in creation scenario and not allowed to changed once set.",
```

Comment:
There's a new way to more officially annotate ARM IDs in service Swagger. I've got a PR out to add it to older AKS API versions [here](https://github.com/Azure/azure-rest-api-specs/pull/19592)

Can you also use it here? I think it should be:
```
      "format": "arm-id",
      "x-ms-arm-id-details": {
        "allowedResources": [
          {
            "type": "Microsoft.Compute/hostGroups"
          }
        ]
      }
```

See [this](https://github.com/Azure/autorest/tree/main/docs/extensions#x-ms-arm-id-details) for more details